src/Core/Parser/Patterns.php

```diff
@@ -241,7 +241,7 @@
 					(?:"(?:\\\"|[^"])*")
 					|(?:\'(?:\\\\\'|[^\'])*\')
 				)
-				|(?P<VariableIdentifier>[a-zA-Z0-9\-_.]*[a-zA-Z]+)  # variable identifiers must contain letters (otherwise they are hardcoded numbers)
+				|(?P<VariableIdentifier>[a-zA-Z0-9\-_.]*[a-zA-Z]+[a-zA-Z0-9.]*) # variable identifiers must contain letters (otherwise they are hardcoded numbers)
```

Comment:
The `\-` should be escaped, also for PHP single quoted strings, or put the dash to the very end of the character class.
Is a dot at the very end really allowed? What about an underscore at the very end?